clicommand/agent_start.go

```diff
@@ -939,6 +948,23 @@ var AgentStartCommand = cli.Command{
 			l.Info("Agents will disconnect after %d seconds of inactivity", agentConf.DisconnectAfterIdleTimeout)
 		}
 
+		// Show allowed environment variables if it is set
+		if len(agentConf.AllowedEnvironmentVariables) > 0 {
+			l.Info("Environment variable allow list is enabled. Only allow-listed environment variables can be set.")
+			l.Debug("Allowed environment variables:")
+
+			for _, v := range agentConf.AllowedEnvironmentVariables {
+				if _, err := regexp.Compile(v); err != nil {
+					l.Fatal("Regex %s in allowed-environment-variables failed to compile: %v", v, err)
+				} else {
+					l.Debug(" - %s", v)
+					if v[0] != '^' || v[len(v)-1] != '$' {
+						l.Warn(" - %s is not anchored! It can result in environment variable leak with partial match.", v)
+					}
```

Comment:
i've come around on this one a bit, but my opinion has evolved to the point that i think that we should either:
- allow whatever, and let customers shoot themselves in the foot if they want - we trust that they know how to write a regex, or:
- disallow any and all regexes not anchored at the start and end, forcing them to fully anchor all regexes

i think the middle ground of warning on what could well be a normal use-case is a bit of a half measure - i'm happy with either of the above options

Reply:
oh also! we can get rid of the `else` above, the program will have exited if there was an error, so being past that line means that there's implicitly no error - see [this passage in Effective Go](https://go.dev/doc/effective_go#if:~:text=In%20the%20Go%20libraries%2C%20you%27ll%20find%20that%20when%20an%20if%20statement%20doesn%27t%20flow%20into%20the%20next%20statement%E2%80%94that%20is%2C%20the%20body%20ends%20in%20break%2C%20continue%2C%20goto%2C%20or%20return%E2%80%94the%20unnecessary%20else%20is%20omitted.)